src/web_app/js/windowport.js

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+/* More information about these options at jshint.com/docs/options */
+
+/* globals trace, chrome */
+/* exported WindowPort */
+
+'use strict';
+
+// This is used to communicate from the Chrome App window to background.js.
+// It opens a Port object to send and receive messages. When the Chrome
+// App window is closed, background.js receives notification and can
+// handle clean up tasks.
+var WindowPort = (function() {
```

Comment:
Unusual (to my eyes at least, but not incorrect). Is there reason not to use the common way of defining new object? Why stuff it in a closure instead? If the goal is to have just one port object can wrap just the port in a closure. It looks like you're trying to add namespace here, but we haven't done that elsewhere in the project.

ie. 

var sendPortMessage;
var addPortMessageListener;
(function() {
  var port = chrome.runtime.connect();
  if (!port) { <report error> }
  sendPortMessage = function(message) {...};
  addPortMessageListener = function(listener) {...};
})();

or if multiple calls to port return the same channel:

var WindowPort = function() {
  this.port_ = chrome.runtime.connect();
};

WindowPort.prototype.sendMessage = function(message) ...
etc